core/math/vector4.h

```diff
@@ -129,28 +128,14 @@ struct _NO_DISCARD_ Vector4 {
 	_FORCE_INLINE_ bool operator<=(const Vector4 &p_vec4) const;
 
 	operator String() const;
+	operator Vector4i() const;
 
 	_FORCE_INLINE_ Vector4() {}
-
-	_FORCE_INLINE_ Vector4(real_t p_x, real_t p_y, real_t p_z, real_t p_w) :
-			x(p_x),
-			y(p_y),
-			z(p_z),
-			w(p_w) {
-	}
-
-	Vector4(const Vector4 &p_vec4) :
-			x(p_vec4.x),
-			y(p_vec4.y),
-			z(p_vec4.z),
-			w(p_vec4.w) {
-	}
-
-	void operator=(const Vector4 &p_vec4) {
-		x = p_vec4.x;
-		y = p_vec4.y;
-		z = p_vec4.z;
-		w = p_vec4.w;
+	_FORCE_INLINE_ Vector4(const real_t p_x, const real_t p_y, const real_t p_z, const real_t p_w) {
```

Comment:
It matches Vector3:
```c++
_FORCE_INLINE_ Vector3() {}
_FORCE_INLINE_ Vector3(const real_t p_x, const real_t p_y, const real_t p_z) {
	x = p_x;
	y = p_y;
	z = p_z;
}
```

Reply:
But we don't add it to simple types, if anything it should be removed from `Vector3`, but adding it to match `Vector3` when it goes against code style isn't good IMO

Reply:
Please propose that we remove it from Vector3 then?

Reply:
I did? That can be done with any other change to that, but having them there doesn't change things here, it's a convention for argument passing and they're *added* in this PR

Reply:
[logfile.txt](https://github.com/godotengine/godot/files/14320258/logfile.txt)

Removing this code causes a few pages of errors.

Reply:
What code? The constant arguments? They weren't in there to begin with so that's not possible

I checked removing *just* the `const` from `Vector4i` and I get no errors at all, did you also remove the copy constructor?

Reply:
Added a PR for cleaning up the broader style issues in these files:
* https://github.com/godotengine/godot/pull/88467

Reply:
Since #88467 has been approved this should be changed